FiskalySDK/FiskalyHttpClient.swift

```diff
@@ -1,22 +1,23 @@
 import Foundation
 import FiskalyClient
 
-func FiskalyClientInvoke(_ request: String) -> String {
-    let resultRaw = _fiskaly_client_invoke(request)
-    let result = String(cString: resultRaw!)
-    _fiskaly_client_free(resultRaw)
-    return result
-}
-
 public class FiskalyHttpClient {
 
     private var context: String
+    private let client: RequestClient
 
     /*
      Initializer
      */
 
-    public init(apiKey: String, apiSecret: String, baseUrl: String) throws {
+    public convenience init(apiKey: String, apiSecret: String, baseUrl: String) throws {
+        try self.init(apiKey: apiKey,
+                      apiSecret: apiSecret,
+                      baseUrl: baseUrl,
+                      client: FiskalyRequestClient())
+    }
+
+    public init(apiKey: String, apiSecret: String, baseUrl: String, client: RequestClient) throws {
```

Comment:
Can't we drop the `convenience` initializer and add a default value to argument `client` instead?
```suggestion
    public init(apiKey: String, apiSecret: String, baseUrl: String, client: RequestClient = FiskalyRequestClient()) throws {
```

Reply:
That looks cleaner. Yes. 

Just reminds me that I still need to add defaults to the request method to remove the overloaded functions. (Thanks for the reminder!)

Reply:
I would have went with that but then the `FiskalyRequestClient` needs to be public as well and is exposed to the outside.

Reply:
This is ok for us. You can adapt the code as suggested.

Reply:
Done, will rebase this branch upon your refactoring branch until EOD.